### ISSUE TITLE
`TypeIs` narrows constrained type variable to `Never`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7760]
### Discussed in https://github.com/microsoft/pyright/discussions/7759

<div type='discussions-op-text'>

<sup>Originally posted by **InSyncWithFoo** April 24, 2024</sup>
Minimal reproducible example ([playground](https://pyright-play.net/?pythonVersion=3.13&strict=true&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigBVEsBJQgKHYBMthzCNCYIABQA3AFxQwAIwBWWAMYwAlFAC0APnqMWAbQEgAumNZRTUAHSX2rLj2BgwOuhKH6ANORTKDoiXRUa9MZm5DxIfPqiSsEhZiBYIlgAhgA2GPAIWFGmAMRQAHIJWCAmsabxMACuIChQIqWmWCmEWDFl8Ymp6YzZUHl0AAL2YA2xFdW19UA)):

```python
from typing_extensions import TypeIs

def is_str(v: object) -> TypeIs[str]:
    ...

def foo[T: (str, int)](v: T) -> T:
    if is_str(v):
        reveal_type(v)  # Never
    return v
```

Is $(str, int) \land str$ really $\emptyset$?</div>

**Comment by rayansostenes** [comment]
I did not know if this is related to the problem I'm having so I created a new issue #7767

**Comment by erictraut** [comment]
This is addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.361, which I just published.
